Feature/history

s8wa2 opened the pull request:
Introduces the ability to retrieve the command last run by cligpt. Using arrows in terminal is pretty natural, there's been countless times using cligpt where I wish I had the ability to slightly modify the last command without copy pasting from the output of the last one.

It's important to note that cligpt must be ran as a function, or called using source, which is why the readme was updated to include steps in doing so.

Please make modifications as needed in the readme, as I'm no wordsmith.

s8wa2 committed:
Add history command to script

s8wa2 committed:
Explain steps required for history to work

Luanf commented:
Hello @s8wa2, thanks for submitting a pull request! I'm glad you've used cligpt countless times :) 

This is a feature I've definitely thought about adding, but had some issues getting it to work myself.
In fact, the code you've submitted doesn't save anything to my history, and I suspect it's because I'm using ZSH.

Are you using Bash?
If you have any means of testing it through ZSH, please let me know too. 

Also, regarding your README changes, I guess I can include an alias command as you did, but just having the /bin added to the PATH should be enough, so maybe I'll expand on the PATH instructions too.

s8wa2 commented:
I am using bash, I tested it using bash and it seemed to work. I did not test it using ZSH which was an oversight. I don't use ZSH, but I can see if I can replicate the history feature for it.

If there's no way to implement a solution for ZSH, perhaps a menu select option (where the user selects either bash or ZSH) would be viable for the install script.

s8wa2 committed:
ZSH compatibility

s8wa2 commented:
Hey @Luanf
I was able to successfully get it working with a fresh zsh / oh-my-zsh install. For zsh compatibility I added a function that checks if zsh is in use then runs the appropriate read / save history commands. 

As for why I added the alias, it's so that history is properly saved. It won't if it is ran as a file instead of a function or from source.

Luanf reviewed (approved):
That's brillant! Certainly codesmith stuff. 

I see what you are saying about the alias now. 
I might tweak the readme a bit sometime soon but I'll merge this already, it's great. 

Thanks for working on this! :nerd_face: 
Here's some :shallow_pan_of_food:  for :thought_balloon: : sometimes I feel like it should also copy the output to the clipboard. But it would also make sense to only do so in the "no" path, where you want to modify something manually. But I'm not sure about this, and what would be preferable to most users. Maybe always copying? Maybe adding a third option "c" ?
